### DISCUSSION TITLE
How to convert rego.ResultSet to json?

### DISCUSSION THREAD

**Discussion by archangel78** [open-policy-agent/community#346] [OPA and Rego]
Can someone explain what the Items in ExpressionValues mean? (Value, Text and Location). I am trying to convert the contents of this value interface to json but i am unable to. Can someone tell me a clean way to convert it to json and extract the Values in it.

`result, err := preparedQry.Eval(ctx, rego.EvalInput(input))`

This is how i am getting the resultset

**Comment by charlieegan3** [comment]
Hi @archangel78, sorry for the delay here.

A result contains a list of Expressions, as I understand it, there is an expression for each query.

e.g. a query like this:

```
data.play; data.play.rule
```

would have two Expressions.

For each expression, there is a value. This is presumably what you want to extract.

Perhaps something like this might might be suitable if you just need a JSON result?


```go
package main

import (
	"context"
	"encoding/json"
	"fmt"
	"github.com/open-policy-agent/opa/rego"
)

func main() {
	ctx := context.Background()

	r := rego.New(
		rego.Module("example.rego", `
package example

foo := { "foo": "bar", "baz": input.x }
`),
		rego.Query("data.example.foo"),
	)

	pq, _ := r.PrepareForEval(ctx)

	input := map[string]interface{}{"x": 2}

	rs, _ := pq.Eval(ctx, rego.EvalInput(input))

	value := rs[0].Expressions[0].Value.(map[string]interface{})

	bs, _ := json.MarshalIndent(value, "", "  ")

	fmt.Println(string(bs))
}
```
